### ISSUE TITLE
FLAKE: github/grpc/run_tests/objc_macos_opt_native/mac-test-basictests[default]

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#22521]
https://source.cloud.google.com/results/invocations/ad79ffd2-eddc-4977-9d5b-46b2b534ca99/targets/github%2Fgrpc%2Frun_tests%2Fobjc_macos_opt_native%2Fmac-test-basictests/tests

Looks like there were 4 test failures here:

```
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testConcurrentRPCsWithErrorsWithV2API]' started.
Thu Mar 26 08:58:50 PDT 2020 - 2020-03-26 08:55:31.589654-0700 xctest[30836:1011362] [Metadata] Got error Connection invalid
Thu Mar 26 08:58:50 PDT 2020 - 2020-03-26 08:55:31.590454-0700 xctest[30836:1011369] DebugSymbols was unable to start a spotlight query: spotlight is not responding or disabled.
Thu Mar 26 08:58:50 PDT 2020 - <unknown>:0: error: -[InteropTestsRemote testConcurrentRPCsWithErrorsWithV2API] : Asynchronous wait failed: Exceeded timeout of 32 seconds, with unfulfilled expectations: "Received trailer for RPC 7", "Received trailer for RPC 5", "Received trailer for RPC 6", "Received trailer for RPC 4", "Received trailer for RPC 3", "Received trailer for RPC 9", "Received trailer for RPC 8", "Received trailer for RPC 2", "Received trailer for RPC 1", "Received trailer for RPC 0".
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testConcurrentRPCsWithErrorsWithV2API]' failed (32.276 seconds).
```

```
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testHijackingInterceptor]' started.
Thu Mar 26 08:58:50 PDT 2020 - /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/objective-c/tests/InteropTests/InteropTests.m:1656: error: -[InteropTestsRemote testHijackingInterceptor] : ((trailingMetadata) == nil) failed: "{
Thu Mar 26 08:58:50 PDT 2020 - }"
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testHijackingInterceptor]' failed (29.555 seconds).
```

```
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testKeepaliveWithV2API]' started.
Thu Mar 26 08:58:50 PDT 2020 - <unknown>:0: error: -[InteropTestsRemote testKeepaliveWithV2API] : Asynchronous wait failed: Exceeded timeout of 32 seconds, with unfulfilled expectations: "Keepalive".
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testKeepaliveWithV2API]' failed (32.060 seconds).
```

```
Thu Mar 26 08:58:50 PDT 2020 - Test Case '-[InteropTestsRemote testLargeUnaryRPC]' started.
Thu Mar 26 08:58:50 PDT 2020 - E0326 08:57:59.022894000 123145344069632 chttp2_transport.cc:2881]     ipv4:74.125.142.81:443: Keepalive watchdog fired. Closing transport.
Thu Mar 26 08:58:50 PDT 2020 - 2020-03-26 08:57:59.023044-0700 xctest[30836:1014333] *** Assertion failure in -[XCTestCase _enqueueFailureWithDescription:inFile:atLine:expected:breakWhenDequeued:](), /Library/Caches/com.apple.xbs/Sources/XCTest/XCTest-13764/Sources/XCTestFramework/Core/XCTestCase.m:399
Thu Mar 26 08:58:50 PDT 2020 - 2020-03-26 08:57:59.023311-0700 xctest[30836:1014333] *** Assertion failure in -[XCTestCase _enqueueFailureWithDescription:inFile:atLine:expected:breakWhenDequeued:](), /Library/Caches/com.apple.xbs/Sources/XCTest/XCTest-13764/Sources/XCTestFramework/Core/XCTestCase.m:399
Thu Mar 26 08:58:50 PDT 2020 - 2020-03-26 08:57:59.030312-0700 xctest[30836:1014333] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Unable to report test assertion failure '((error.code) equal to (GRPC_STATUS_UNAVAILABLE)) failed: throwing "Unable to report test assertion failure '((error.code) equal to (GRPC_STATUS_UNAVAILABLE)) failed: ("1") is not equal to ("14") - Received status 1 instead of UNAVAILABLE (14).' from /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/objective-c/tests/InteropTests/InteropTests.m:1405 because it was raised inside test case -[InteropTestsRemote testKeepaliveWithV2API] which has no associated XCTestRun object. This may happen when test cases are constructed and invoked independently of standard XCTest infrastructure, or when the test has already finished." - Received status 1 instead of UNAVAILABLE (14).' from /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/objective-c/tests/InteropTests/InteropTests.m:1405 because it was raised inside test case -[InteropTestsRemote testKeepaliveWithV2API] which has no associated XCTestRun object. This may happen when test cases are constructed and invoked independently of standard XCTest infrastructure, or when the test has already finished.'
Thu Mar 26 08:58:50 PDT 2020 - *** First throw call stack:
Thu Mar 26 08:58:50 PDT 2020 - (
Thu Mar 26 08:58:50 PDT 2020 - 	0   CoreFoundation                      0x00007fff548eb66b __exceptionPreprocess + 171
Thu Mar 26 08:58:50 PDT 2020 - 	1   libobjc.A.dylib                     0x00007fff7bb3d942 objc_exception_throw + 48
Thu Mar 26 08:58:50 PDT 2020 - 	2   CoreFoundation                      0x00007fff548f1252 +[NSException raise:format:arguments:] + 98
Thu Mar 26 08:58:50 PDT 2020 - 	3   Foundation                          0x00007fff56a9fa0b -[NSAssertionHandler handleFailureInFunction:file:lineNumber:description:] + 165
Thu Mar 26 08:58:50 PDT 2020 - 	4   XCTest                              0x0000000105ec81a8 -[XCTestCase _enqueueFailureWithDescription:inFile:atLine:expected:breakWhenDequeued:] + 1034
Thu Mar 26 08:58:50 PDT 2020 - 	5   XCTest                              0x0000000105ed20c0 _XCTFailureHandler + 408
Thu Mar 26 08:58:50 PDT 2020 - 	6   MacTests                            0x0000000108fd3a15 __38-[InteropTests testKeepaliveWithV2API]_block_invoke + 3045
Thu Mar 26 08:58:50 PDT 2020 - 	7   MacTests                            0x0000000108fff4fc -[InteropTestsBlockCallbacks didCloseWithTrailingMetadata:error:] + 124
Thu Mar 26 08:58:50 PDT 2020 - 	8   MacTests                            0x000000010922b4ac __61-[GRPCStreamingProtoCall didCloseWithTrailingMetadata:error:]_block_invoke + 135
Thu Mar 26 08:58:50 PDT 2020 - 	9   libdispatch.dylib                   0x00007fff7c7225fa _dispatch_call_block_and_release + 12
Thu Mar 26 08:58:50 PDT 2020 - 	10  libdispatch.dylib                   0x00007fff7c71adb8 _dispatch_client_callout + 8
Thu Mar 26 08:58:50 PDT 2020 - 	11  libdispatch.dylib                   0x00007fff7c72f217 _dispatch_queue_serial_drain + 635
Thu Mar 26 08:58:50 PDT 2020 - 	12  libdispatch.dylib                   0x00007fff7c722166 _dispatch_queue_invoke + 373
Thu Mar 26 08:58:50 PDT 2020 - 	13  libdispatch.dylib                   0x00007fff7c72f07a _dispatch_queue_serial_drain + 222
Thu Mar 26 08:58:50 PDT 2020 - 	14  libdispatch.dylib                   0x00007fff7c722166 _dispatch_queue_invoke + 373
Thu Mar 26 08:58:50 PDT 2020 - 	15  libdispatch.dylib                   0x00007fff7c72ff0d _dispatch_root_queue_drain_deferred_wlh + 332
Thu Mar 26 08:58:50 PDT 2020 - 	16  libdispatch.dylib                   0x00007fff7c733d21 _dispatch_workloop_worker_thread + 880
Thu Mar 26 08:58:50 PDT 2020 - 	17  libsystem_pthread.dylib             0x00007fff7ca6bfd2 _pthread_wqthread + 980
Thu Mar 26 08:58:50 PDT 2020 - 	18  libsystem_pthread.dylib             0x00007fff7ca6bbe9 start_wqthread + 13
Thu Mar 26 08:58:50 PDT 2020 - )
Thu Mar 26 08:58:50 PDT 2020 - libc++abi.dylib: terminating with uncaught exception of type NSException
Thu Mar 26 08:58:50 PDT 2020 -
Thu Mar 26 08:58:50 PDT 2020 - Restarting after unexpected exit or crash in InteropTestsRemote/testLargeUnaryRPC; summary will include totals from previous launches.
```

**Comment by muxi** [comment]
This has been fixed by #22687

**Comment by yashykt** [comment]
https://source.cloud.google.com/results/invocations/1494e9c0-e279-4db7-8800-a00f5fd932fb/targets/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_objc_macos_opt_native/tests

```
Wed Jul 22 16:19:09 PDT 2020 - Test Case '-[InteropTestsRemote testHijackingInterceptor]' failed (32.259 seconds).
```
saw this test fail

**Comment by markdroth** [comment]
Just saw this failure again:

```
Fri Aug 21 12:02:34 PDT 2020 - Test Case '-[InteropTestsRemote testConcurrentRPCsWithErrorsWithV2API]' started.
Fri Aug 21 12:02:34 PDT 2020 - 2020-08-21 12:02:06.415344-0700 xctest[33243:64929] [Metadata] Got error Connection invalid
Fri Aug 21 12:02:34 PDT 2020 - 2020-08-21 12:02:06.415964-0700 xctest[33243:65056] DebugSymbols was unable to start a spotlight query: spotlight is not responding or disabled.
Fri Aug 21 12:02:34 PDT 2020 - <unknown>:0: error: -[InteropTestsRemote testConcurrentRPCsWithErrorsWithV2API] : Asynchronous wait failed: Exceeded timeout of 32 seconds, with unfulfilled expectations: "Received trailer for RPC 2", "Received trailer for RPC 5", "Received trailer for RPC 7", "Received trailer for RPC 4", "Received trailer for RPC 8", "Received trailer for RPC 9", "Received trailer for RPC 0", "Received trailer for RPC 6", "Received trailer for RPC 3", "Received trailer for RPC 1".
Fri Aug 21 12:02:34 PDT 2020 - Test Case '-[InteropTestsRemote testConcurrentRPCsWithErrorsWithV2API]' failed (32.299 seconds).
```

**Comment by markdroth** [comment]
Another failure:

```
Tue Dec 15 13:51:31 PST 2020 - Test Case '-[InteropTestsRemote testHijackingInterceptor]' started.
Tue Dec 15 13:52:04 PST 2020 - /Volumes/BuildData/tmpfs/src/github/grpc/workspace_objc_macos_opt_native/src/objective-c/tests/InteropTests/InteropTests.m:1710: error: -[InteropTestsRemote testHijackingInterceptor] : Asynchronous wait failed: Exceeded timeout of 32 seconds, with unfulfilled expectations: "Internal gRPC call completed.".
Tue Dec 15 13:52:04 PST 2020 - Test Case '-[InteropTestsRemote testHijackingInterceptor]' failed (32.298 seconds).
```

**Comment by yulin-liang** [comment]
Fixed: #25958
